lua/common.lua

```diff
@@ -42,3 +41,72 @@ end
 function wesnoth.wml_actions.clear_chat(cfg)
 	wesnoth.interface.clear_chat_messages()
 end
+
+---
+-- Get current game zoom level.
+--
+-- [get_zoom]
+--     variable=zoom
+-- [/get_zoom]
+---
+function wesnoth.wml_actions.get_zoom(cfg)
+	local varname = cfg.variable or "zoom"
+	wml.variables[varname] = wesnoth.interface.zoom(1, true)
+end
+
+function wesnoth.wml_actions.fading_message(cfg)
+	local message = cfg.message
+	local delay_time = tonumber(cfg.time)
+	local fade_time = tonumber(cfg.fade_time) or 100
+	local matches = wesnoth.units.find_on_map(cfg)
+	if matches ~= nil then
+		local name = matches[1].name
+		local unit_x = matches[1].x
+		local unit_y = matches[1].y
+		wesnoth.interface.highlight_hex(unit_x, unit_y)
+		local options = {}
+		options.max_width = "80%"
+		options.color = "FFFFFF"
+		options.bgcolor = "000000"
+		options.duration = delay_time - fade_time
+		options.fade_time = fade_time
+		options.halign="center"
+		options.valign="bottom"
+		options.location={0,100}
+		local handle = wesnoth.interface.add_overlay_text("<span size='x-large'>" .. name .. "</span>\n<span size='large'>" .. message .. "</span>", options)
+		wesnoth.interface.delay(delay_time)
+		wesnoth.interface.deselect_hex()
+	end
+end
+
+---
+-- misc functions from EoMA, primarily used to facilitate custom abilities for elementals
+---
+function wesnoth.wml_actions.remove_array_duplicates(cfg)
+	local name = cfg.name
+	local attribute = cfg.attribute
+
+	local values = {}
+	local inArray = wml.array_access.get(name)
+	local outArray = {}
+	for _, u in pairs(inArray) do
+		if values[u[attribute]] then
+			-- skip duplicate
```

Comment:
I think the `luacheck` warning here is legitimate; can't there at least be *something* that goes here?

Reply:
Reversed the logic here which should resolve this.

Reply:
...ok, I see this is resolved in [f302866](https://github.com/cooljeanius/Flight_Freedom/pull/22/commits/f3028668141506dff6811cc60220072059aaf982).